### ISSUE TITLE
WebGPU requires timestamps in nanoseconds

### ISSUE THREAD

**Issue by Draradech** [gfx-rs/wgpu#6719]
WebGPU requires timestamps in nanoseconds (https://gpuweb.github.io/gpuweb/#timestamp)
wgpu requires multiplication with get_timestamp_period (https://docs.rs/wgpu/latest/wgpu/enum.QueryType.html#variant.Timestamp)

This results in Firefox (135.0a1 (2024-12-11)) reporting wrong values on devices which don't use nanosecond resolution for their timestamps (like Intel Iris Xe)
Chart lists times (in ms) as wallclock / ts-query:
![grafik](https://github.com/user-attachments/assets/09c4d55d-99cf-4ad8-80b5-9385c8fb22e2)

Test code (web and wgpu-py) is available here: https://glitch.com/edit/#!/rowan-pinnate-yacht

For reference: https://github.com/gfx-rs/wgpu/issues/3741

**Comment by Wumpf** [comment]
Firefox not providing the correct values is somewhat orthogonal arguably. Yes, as it stands Firefox has to implement a compute pass to convert the timestamp on-gpu when implementing WebGPU using wgpu since that's how wgpu is specified today.
However, I think we want at least be able to allow to avoid this additional overhead of gpu sided timestamp conversion. But likely that should be opt-in rather opt-out, otherwise it's confusing for anyone who is reasonably expecting wgpu to behave like the WebGPU api in that regard! 🤔

**Comment by cwfitzgerald** [comment]
Yeah native should be off, using the timestamp period, and Firefox will turn it on in core.

Still need to implement it in wgpu though.

**Comment by Draradech** [comment]
Does Firefox still need to do something on their side? I still get timestamps in device resolution (in Firefox 149.0, Intel Iris Xe, both Linux Mint and Windows 11).

**Comment by ErichDonGubler** [comment]
Huh, it looks like there is. I don't think we collectively realized there was still an opt-in step that needed to be applied (sorry for dropping that ball for your hard work, @cwfitzgerald!). I'm following up in Firefox's [bug 1986745](https://bugzilla.mozilla.org/show_bug.cgi?id=1986745).

**Comment by ErichDonGubler** [comment]
@Draradech: FYI: I've reached out to you on ☝🏻 this bug with some instructions for trying out a build.
